recipes/wtl/all/conanfile.py

```diff
@@ -0,0 +1,24 @@
+from conans import ConanFile, tools
+
+import os
+
+class WTLConan(ConanFile):
+    name = "wtl"
+    license = "Microsoft Public License"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/wtl"
+    description = "Windows Template Library (WTL) is a C++ library for developing Windows applications and UI components."
+    topics = ("atl", "template library", "windows", "template", "ui", "gdi")
+
+    settings = {'os': ['Windows']}
```

Comment:
Raise `ConanInvalidConfiguration` instead of constraining the settings:
https://github.com/conan-io/conan-center-index/pull/841#issuecomment-585160197

Reply:
I preferred an approach with constraining settings, because I read [the guideline](https://docs.conan.io/en/latest/mastering/conditional.html) and in case with WTL, the recipe is never supposed to work with os other than Windows. Moreover, the declarative approach `settings = {'os': ['Windows']}` is more readable and does not rely on user defined logic or error messages written by a user. A declarative constraint can be handled by conan client in a unified way. And since [the question](https://github.com/conan-io/conan/issues/6534) has not been resolved yet, maybe it is better to leave things as is and correct such constraints when guidelines are elaborated?

Reply:
Indeed, both ways have their merits.

Reply:
yes, for now let's allow `setting = {'os': ['Windows']}` construct.
in the future, once we make a decision, we may disallow syntax in hooks (or maybe even in conan itself), then we will fix recipe(s) affected.
for now, it's fully legitim.

Reply:
We will have to set a policy about this. I will open an issue about it